boa/src/builtins/object/mod.rs

```diff
@@ -109,6 +119,94 @@ impl Object {
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor
+    pub fn get_own_property_descriptor(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap().to_object(ctx)?;
+        if let Some(key) = args.get(1) {
+            let key = key.to_property_key(ctx)?;
+            let desc = object.borrow().get_own_property(&key);
+            Self::from_property_descriptor(desc, ctx)
+        } else {
+            Ok(Value::undefined())
+        }
+    }
+
+    /// `Object.getOwnPropertyDescriptors( object )`
+    ///
+    /// Returns all own property descriptors of a given object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/10.0/index.html#sec-object.getownpropertydescriptor
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors
+    pub fn get_own_property_descriptors(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap_or(&Value::undefined()).to_object(ctx)?;
+        let descriptors = ctx.construct_object();
+
+        for key in object.borrow().keys() {
+            let desc = object.borrow().get_own_property(&key);
+            let descriptor = Self::from_property_descriptor(desc, ctx)?;
+
+            if !descriptor.is_undefined() {
+                descriptors
+                    .borrow_mut()
+                    .insert(key, Property::data_descriptor(descriptor, Attribute::all()));
+            }
+        }
+
+        Ok(Value::Object(descriptors))
+    }
+
+    /// https://www.ecma-international.org/ecma-262/10.0/index.html#sec-frompropertydescriptor
+    fn from_property_descriptor(desc: Property, ctx: &mut Context) -> Result<Value> {
+        let mut descriptor = ObjectInitializer::new(ctx);
+
+        if let Some(value) = &desc.value {
+            descriptor.property("value", value, Attribute::all());
+        }
+        if let Some(set) = &desc.set {
+            descriptor.property("set", set, Attribute::all());
+        }
+        if let Some(get) = &desc.get {
+            descriptor.property("get", get, Attribute::all());
+        }
+
+        Ok(Value::Object(
+            descriptor
+                .property("writable", Value::from(desc.writable()), Attribute::all())
+                .property(
+                    "enumerable",
+                    Value::from(desc.enumerable()),
+                    Attribute::all(),
+                )
+                .property(
+                    "configurable",
+                    Value::from(desc.configurable()),
+                    Attribute::all(),
+                )
+                .build(),
+        ))
```

Comment:
```suggestion
		descriptor
                .property("writable", Value::from(desc.writable()), Attribute::all())
                .property(
                    "enumerable",
                    Value::from(desc.enumerable()),
                    Attribute::all(),
                )
                .property(
                    "configurable",
                    Value::from(desc.configurable()),
                    Attribute::all(),
                );

        Ok(descriptor.build().into())
```
its cleaner if we put property builting out of `Ok()` and use `into()` (BTW this needs rustfmt)